src/lava/lib/optimization/problems/problems.py

```diff
@@ -103,7 +103,7 @@ def constraints(self):
         return None
 
     def evaluate_cost(self, solution: np.ndarray) -> int:
-        return solution.T @ self._q_cost.coefficients[2] @ solution
+        return int(self._q_cost(solution))

```

Comment:
self._q_cost is of type Cost. But in class Cost, I cannot see a __call__ function which would support this call. Am I missing something?

Apart from that, I like the idea that class Cost can calculate the cost of a specific solution.

Reply:
Cost is inheriting from CoefficientTensorsMixin that implements the __call__ method, no need to reimplement it in Cost.

Reply:
Oh, I missed that inheritance in cost.